docs/topics/hetzner-baremetal.md

```diff
@@ -0,0 +1,377 @@
+
+### Creating new user in Robot 
+To create new user in Robot, click on the `Create User` button in the Hetzner Robot console. Once you create the new user, a user ID will be provided to you via email from Hetzner Robot. The password will be the same that you used while creating the user.
+
+![robot user](../pics/robot_user.png)
+
+This is a required for following the next step.
+
+### creating and verify ssh-key in hcloud 
+```bash
+hcloud ssh-key create --name test --public-key-from-file ~/.ssh/test.pub
+# to verify use 
+hcloud ssh-key list
+```
+This is an important step as the ssh key is used to access the servers. Make sure you are using the correct ssh key name.
+
+The `test` is the name of the ssh key that we have created above. It is because the generated manifest references `test` as the ssh key name. 
+```yaml
+  sshKeys:
+    hcloud:
+    - name: test
+```
+
+### Create a secret for Hetzner (Hcloud + Robot)
+
+In order for the provider integration hetzner to communicate with the Hetzner API ([HCloud API](https://docs.hetzner.cloud/) + [Robot API](https://robot.your-server.de/doc/webservice/en.html#preface)), we need to create a secret with the access data. The secret must be in the same namespace as the other CRs.
+
+```shell
+export HCLOUD_TOKEN="<YOUR-TOKEN>" \
+export HETZNER_ROBOT_USER="<YOUR-ROBOT-USER>" \
+export HETZNER_ROBOT_PASSWORD="<YOUR-ROBOT-PASSWORD>" \
+export HETZNER_SSH_PUB_PATH="<YOUR-SSH-PUBLIC-PATH>" \
+export HETZNER_SSH_PRIV_PATH="<YOUR-SSH-PRIVATE-PATH>"
+```
+
+- HCLOUD_TOKEN: The project where your cluster will be placed. You have to get a token from your HCloud Project.
+- HETZNER_ROBOT_USER: The User you have defined in Robot under settings/web.
+- HETZNER_ROBOT_PASSWORD: The Robot Password you have set in Robot under settings/web.
+- HETZNER_SSH_PUB_PATH: The Path to your generated Public SSH Key.
+- HETZNER_SSH_PRIV_PATH: The Path to your generated Private SSH Key. This is needed because CAPH uses this key to provision the node in Hetzner Dedicated.
+
+```shell
+kubectl create secret generic hetzner --from-literal=hcloud=$HCLOUD_TOKEN --from-literal=robot-user=$HETZNER_ROBOT_USER --from-literal=robot-password=$HETZNER_ROBOT_PASSWORD
+
+kubectl create secret generic robot-ssh --from-literal=sshkey-name=test --from-file=ssh-privatekey=$HETZNER_SSH_PRIV_PATH --from-file=ssh-publickey=$HETZNER_SSH_PUB_PATH
+```
+
+> NOTE: sshkey-name should must match the name that is present in hetzner otherwise the controller will not know how to reach the machine.
+
+Patch the created secrets so that they get automatically moved to the target cluster later. The following command helps you do that:
+
+```shell
+kubectl patch secret hetzner -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+kubectl patch secret robot-ssh -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
+```
+
+The secret name and the tokens can also be customized in the cluster template.
```

Comment:
doesn't the cluster object have some pre-conditions on how the ssh keys are named? Currently, this is not described. Probably we have hard-coded it on "test" right?

Reply:
We've hardcoded it in the template.

Reply:
so mention it in the docs ;)